inductiva/scenarios.py

```diff
@@ -19,11 +20,15 @@ def create_input_files(self, simulator: Simulator, input_dir: Path):
         """To be implemented in subclasses."""
         pass
 
-    def read_commands_from_file(self, commands_path: str):
+    def read_commands_from_file(self, commands_file: Union[str, io.StringIO]):
         "Read list of commands from commands.json file"
-        with open(commands_path, "r", encoding="utf-8") as f:
-            commands = json.load(f)
-        return commands
+        if isinstance(commands_file, str):
```

Comment:
This if could just read the file and return be just in the end.

Reply:
But since this is using the context manager to open/close the file, it wouldn't look that well because afterwards we would need to write the conditional logic again to know that if the file was opened here, we need to close it